Спринт 10

cyberschoolinmoscow opened the pull request:
1) Добавлены разрешения в манифест
2) Созданы классы описывающие элемент списка
3) В SearchActivity добавлен RecyclerView для просмотра списка треков
4) Сверстана карточка трека

cyberschoolinmoscow committed:
1) Созданы классы, описывающие элемент списка, Track, TrackViewHolder и TrackAdapter.
2) Для загрузки обложки трека используется Glide.
3) Элементы списка сверстаны.

cyberschoolinmoscow committed:
1) Добавлены разрешения в манифест

cyberschoolinmoscow committed:
Исправлен размер corner_radius_pic=4dp, и изменен размер скругления в customRoundedImageView.

faritowich commented on app/src/main/java/com/practicum/playlistmaker/TrackViewHolder.kt:
```diff
@@ -0,0 +1,34 @@
+        trackName = itemView.findViewById(R.id.trackName)
+        artistName = itemView.findViewById(R.id.artistName)
+        trackTime = itemView.findViewById(R.id.trackTime)
+        artistImage = itemView.findViewById(R.id.artistImage)
```
Можно и без блока инит обойтись:)
`private val trackName: TextView = itemView.findViewById(R.id.trackName)`

cyberschoolinmoscow replied:
Спасибо, убрала

faritowich commented on app/src/main/java/com/practicum/playlistmaker/TrackViewHolder.kt:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    fun bind(model: Track) {
+        trackName.text = "#${model.trackName}"
```
А зачем тут решетка?

cyberschoolinmoscow replied:
Спасибо, убрала. Скорее всего из теории на платформе скопировала.

faritowich commented on app/src/main/res/layout/track_cardview.xml:
```diff
@@ -0,0 +1,77 @@
+                style="@style/text_main"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center_vertical" />
```
Здесь и далее: для удобства визуального восприятия верстки в TextView удобно использовать атрибут `tools:text="Track Name"`
Он позволит указать текста в лейауте только для разработчика, при этом в приложении их не будет. 
Тогда не будет пустого TextView на превью, как сейчас, а будет видны тексты.
Чтобы использовать его, надо добавить в корневой лейаут строку:
    `xmlns:tools="http://schemas.android.com/tools"`

cyberschoolinmoscow replied:
Спасибо, добавила

faritowich commented on app/src/main/res/layout/track_cardview.xml:
```diff
@@ -0,0 +1,77 @@
+            android:layout_marginHorizontal="18dp" />
+
+    </LinearLayout>
+</androidx.cardview.widget.CardView>
```
Можно и без CardView обойтись, а закругления сделать средствами Glide. Так тоже можно, но имхо это лишняя вложенность и усложнение хмл-файла

cyberschoolinmoscow replied:
Спасибо, в теории на платформе был пример с cardview, можно его убрать, но я подумала, что так надо.

faritowich commented:
<img width="417" alt="Screenshot 2024-05-19 at 12 06 05" src="https://github.com/cyberschoolinmoscow/PlaylistMaker/assets/64851346/0bdbbd8d-6abc-4bbb-ba84-90a761d67970">

cyberschoolinmoscow committed:
Исправлено согласно замечаниям:
1) Удален блок init у класса TrackViewHolder.
2) Исправлено название трека в классе TrackViewHolder.
3) В TextView добавлен атрибут tools:text в верстке track_cardview
4) Устранена лишняя полоса под списком песен в верстке track_cardview
